File: js/0118.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
    if (numRows <= 0) return []
    var res = [[1]]
    for (let i = 1; i < numRows; i++) {
        res.push([1])
        for (let j = 1; j < i; j++) {
            res[i].push(res[i - 1][j - 1] + res[i - 1][j])
        }
        res[i].push(1)
    }
    return res
};

File: py/0118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = [[1] for _ in range(numRows)]
        for i in range(1, numRows):
            for j in range(1, i):
                res[i].append(res[i-1][j]+res[i-1][j-1])
            res[i].append(1)
        return res
